src/bun.js/bindings/bindings.zig

```diff
@@ -4615,6 +4639,12 @@ pub const JSValue = enum(JSValueReprInt) {
     pub inline fn isObject(this: JSValue) bool {
         return this.isCell() and this.jsType().isObject();
     }
+    pub inline fn isArray(this: JSValue) bool {
+        return this.isCell() and this.jsType().isArray();
+    }
+    pub inline fn isFunction(this: JSValue) bool {
+        return this.isCell() and this.jsType().isFunction();
```

Comment:
note that there is a distinction between is it a function, and is it callable

a proxy is not a function

but it might be callable.